native/cocos/core/scene-graph/Node.h

```diff
@@ -657,7 +657,7 @@ class Node : public CCObject {
     // NOTE: TypeArray created in node.jsb.ts _ctor should have the same memory layout
     uint32_t _eventMask{0};                                             // Uint32: 0
     uint32_t _layer{static_cast<uint32_t>(Layers::LayerList::DEFAULT)}; // Uint32: 1
-    uint32_t _transformFlags{0};                                        // Uint32: 2
+    uint32_t _transformFlags{static_cast<uint32_t>(TransformBit::TRS)}; // Uint32: 2
```

Comment:
` this._transformFlags |= TransformBit.TRS;` was set in `_onBatchCreated`.

If we modify the initialized value to TRS, will it mark it as dirty twice and call `updateWorldTransform` twice?

For instance,
```ts
const scene = deserialize(archive) as Scene;
const node = scene.getChildByPath('Node')!;
const worldScale = node.worldScale;  // call updateWorldTransform internally, and reset child._transformFlags = TransformBit.NONE;

// Use worldScale 

director.runScene(scene); // call scene._load -> scene._onBatchCreated -> node._onBatchCreated -> this._transformFlags |= TransformBit.TRS; 
//  this._transformFlags is set to TransformBit.TRS again, so updateWorldTransform will be invoked twice, but in fact, the world transform was not changed at all.

```

Reply:
I'd recommend deleting transform flags modifying in `Node._onBatchCreated`.

Reply:
That's ok for me.

Reply:
https://github.com/cocos/cocos-engine/pull/16353/commits/abb7ea2d65960b25ca3f3e0a5a72bc5514874fd3